admin.php

```diff
@@ -560,8 +592,9 @@ private function printPageStats($page, $match) {
         $this->ptln('<div class="stats">', +2);
 
         if ($this->hasApplicableMatches($page)) {
+            $checkedAll = $this->hasCheckedAll($page);
             $this->ptln('<span class="apply" title="' . $this->getLang('ttl_applyfile') . '">', +2);
-            $this->ptln('<input type="checkbox" id="' . $page . '" />');
+            $this->ptln('<input class="check-full-page" type="checkbox" ' . ($checkedAll ? 'checked ' : '') . 'id="' . $page . '" />');
```

Comment:
I'm not sure follow what you are trying to do here. It looks like you try to preserve marked matches over multiple previews. I see that it can be useful when you are refining replacement string. But when you change regexp you will end up with a different set of matches. Some of them may have the same page offset, some will have another, so only part of marks will be preserved. And even for those that have survived regexp change you might want to double check if they are still placed correctly. You could reset the marks in regexp change, but then Preview button behavior would be inconsistent -- sometimes it resets marks, sometimes it doesn't. I think it's safer to let the user to polish regexp and replacement strings first, and then do mark&apply.

And again, this has nothing to do with max_input_vars. Should be a separate (rejected ;) PR.

Reply:
> matches are identified by their exact offsets actually makes this quite safe.

It doesn't. Offset+length would be safe-ish. Even then groups within regexp could match different things while keeping the total length the same. Offset on it's own is no good at all. Taken to extreme, match at offset 0 could be a first letter on a page or the whole page.

> But maybe in that case it would also be cool to...

That's exactly what I'm talking about. You introduce some very fragile feature and then begins the game of propping it from all sides to ensure it works properly and doesn't do any damage.

> Maybe an Admin setting

I plan to add more features to UI (hence the "massive refactoring") including some persistent settings. I'm still not convinced that this feature should be added. But at least settings are coming :)

Reply:
> It doesn't. Offset+length would be safe-ish. Even then groups within regexp could match different things while keeping the total length the same.

That's a pathological borderline case. Users should be circumspect enough to take care what they are doing.
If they are ticking the checkmark and leaving it in place but then go on to change the regex to match and replace something completely different at the same offset with the same length (or even only at the same offset), this is quite a coincidence and probably intended.

> That's exactly what I'm talking about. You introduce some very fragile feature and then begins the game of propping it from all sides to ensure it works properly and doesn't do any damage.

I strongly disagree with your security paranoia here. As far as I'm concerned the way it works in my fork atm seems quite solid to me, without much possibility to easily run into accidents.

All checkmarks are unchecked at the beginning. If the user checks one he should be aware that he checked it, and double-check later when he changed the regex in a way that is still finding a match at the same offset that he already checked and left checked. And the warning when some previously checked matches (identified by their offsets) don't exist anymore in the new result set would only be a nice add-on IMO, nothing crucial at all. Users should be able to take care what they are doing without much [parental guidance](https://www.youtube.com/watch?v=rTcSMUXyicE) IMO. One would need to cover the whole world in wad to prevent inattentive people from making mistakes.